Kitodo-API/src/main/java/org/kitodo/config/Config.java

```diff
@@ -11,141 +11,197 @@
     private static final String CONFIG_FILE = "kitodo_config.properties";
-    private static final String METADATA_DIRECTORY = "MetadatenVerzeichnis";
-    private static final String CONFIG_DIR = "KonfigurationVerzeichnis";
+    public static final int INT_PARAMETER_NOT_DEFINED_OR_ERRONEOUS = 0;
+
+    public static final String DIR_MODULES = "moduleFolder";
 
     /**
-     * Get Kitodo data directory.
+     * Absolute path to the directory that process directories will be created
+     * in, terminated by a directory separator ("/"). The servlet container must
+     * have write permission to that directory.
+     */
+    private static final String DIR_PROCESSES = "MetadatenVerzeichnis";
+
+    /**
+     * Absolute path to the directory that the configuration files are stored
+     * in, terminated by a directory separator ("/").
      *
-     * @return String
+     * <p>
+     * Note: Several, but not all configuration files are read from that
+     * directory. You may want to decide to point this path to the directory
+     * where the servlet container will extract the configuration files to (like
+     * webapps/kitodo/WEB-INF/classes) in order to make sure they are found.
+     */
+    private static final String DIR_XML_CONFIG = "KonfigurationVerzeichnis";
+
+    /**
+     * Returns the directory that contains the process directories.
+     *
+     * @return the directory for the process directories
      */
     public static String getKitodoDataDirectory() {
-        return getParameter(METADATA_DIRECTORY);
+        return getParameter(DIR_PROCESSES);
     }
 
     /**
-     * Get Kitodo config directory.
+     * Returns the directory that contains XML configuration files.
      *
-     * @return String
+     * @return the directory for XML configuration files
      */
     public static String getKitodoConfigDirectory() {
-        return getParameter(CONFIG_DIR);
+        return getParameter(DIR_XML_CONFIG);
     }
 
     /**
-     * Request selected parameter from configuration.
+     * Returns the selected parameter from the configuration file. Throws a
+     * {@code NoSuchElementException} if no such parameter exists.
      *
-     * @return Parameter as String
+     * @param key
+     *            key whose value is to be returned
+     * @return value for the requested key
+     * @throws NoSuchElementException
+     *             if parameter taken from config file is null or exception
+     *             occurred
      */
-    public static String getParameter(String parameter) {
+    public static String getParameter(String key) {
         try {
-            return getConfig().getString(parameter);
+            return getConfig().getString(key);
         } catch (NoSuchElementException e) {
-            throw new NoSuchElementException(
-                    "No configuration found in kitodo_config.properties for key " + parameter + "!");
+            logger.catching(e);
```

Comment:
Logger and rethrow in one black? IS it necessary?

Reply:
This was there already. I only added logging of caught exception. Because as to coding guidelines, exception being caught must be logged. Btw., the exception is *not rethrown*, but instead a *new equal* exception is thrown.

Reply:
That was my question. If it is necessary to add there logging if in any case we throw exception there  - the same but with more meaningful message.

Reply:
Coding guideline says that it should always be logged.

Reply:
Not when it is throw further.

Reply:
But it is not thrown further. A new exception is thrown. Thats something different, isn't it?

Reply:
It is thrown further, maybe use:
throw new NoSuchElementException( "No configuration found in kitodo_config.properties for key " + parameter + "!", e);
to make it more clear.

Reply:
Correct solution should be to change the message in the existing exception.